### PR TITLE
Fix 5455 updating demo environments to vite

### LINKED ISSUE #5455: Updating demo environments to Vite

### PR DESCRIPTION
**Pull request by nitrogenous**
Fix #5455 updating demo environments to vite

### CODE REVIEW

File: components/doc/common/codeeditor/index.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+            '@vitejs/plugin-react': '^4.0.0'
+        };
+        const { files, dependencies, sourceFileName } = getVite({ dependencies: sandboxDependencies, ...props }, sourceType);
 
         files['sandbox.config.json'] = {
             content: {
-                infiniteLoopProtection: false
+                infiniteLoopProtection: false,
+                container: {
+                    node: '16'
```
Comment (melloware) [inline components/doc/common/codeeditor/index.js:16]: Should we use Node 18 or 20 since those are LTS now?

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/doc/common/codeeditor/index.js (hunk 1 above)
Comment (nitrogenous) [reply components/doc/common/codeeditor/index.js:16]: Codesandbox not supporting 18 or 20. Unfortunately this is the lates version that we can use. This is why we are also using the vite 4.0.0 for codesandbox

Comment (nitrogenous) [reply components/doc/common/codeeditor/index.js:16]: When i set the node version higher than 16 it sets the node version to the default value which is 10


image

Comment (nitrogenous) [reply components/doc/common/codeeditor/index.js:16]: Codesandbox have this buckets called DevBox and it supports Node 18 or 20 we can basically use whatever we want but I couldn't find any examples about creating DevBox buckets via API

Comment (melloware) [reply components/doc/common/codeeditor/index.js:16]: OK no problem thanks for checking!

**Review by melloware — APPROVED** [review]

**Review by gucal — APPROVED** [review]

**Pull request merged**